### ISSUE TITLE
Support header "pinning" progress percentage

### ISSUE THREAD

**Issue by VasylPedos** [letsar/flutter_sticky_header#29]
Hi!

I wonder if it is possible to implement some kind of animation when the header is about to be pinned like it is done in the second example of [sticky_headers](https://pub.dev/packages/sticky_headers) library:
![demo2](https://user-images.githubusercontent.com/40064585/62283290-e777fc00-b459-11e9-8c5d-eefb5be39c6c.gif)

As I can understand `scrollPercentage` property of `SliverStickyHeaderState` changes only when the header scrolls off the viewport. So it would be nice to provide one more property indicating the progress of "pushing out" the previous header or something like this.

Thanks!

**Comment by JWong1105** [comment]
Yes, I also think so... this kind of design seems more important than the pinned state. The users wanna see how the header will change when it is about to be pinned. Thank u.

**Comment by letsar** [comment]
Good idea, but I'm not sure I can do it with the current state of this project. A header should know the state of the previous one in order to do this kind of feature.

**Comment by JWong1105** [comment]
您的邮件我已收到！祝您天天好心情~王杰
